thavalon-server/src/game/state/on_mission.rs

```diff
@@ -118,6 +119,36 @@ impl GameState<OnMission> {
     fn includes_player(&self, player: &str) -> bool {
         self.proposal().players.contains(player)
     }
+
+    // Append messages for lovers to an exisiting list of effects. This will not modify effects if there are no lovers in game,
+    // so this is safe to call in any game.
+    fn add_lover_effects(&self, effects: &mut Vec<Effect>) {
+        let tristan_display_name = self.game.display_name_from_role(Role::Tristan);
+        let tristan_on_mission = tristan_display_name.map_or(None, |x| Some(self.proposals.get(self.phase.proposal_index).unwrap().players.contains(x)));
+        let iseult_display_name = self.game.display_name_from_role(Role::Iseult);
+        let iseult_on_mission = iseult_display_name.map_or(None, |x| Some(self.proposals.get(self.phase.proposal_index).unwrap().players.contains(x)));
+        // First, add messages going to Tristan. If there is no Tristan, do nothing.
+        if tristan_display_name.is_some() {
+            effects.push(self.build_lover_effect(&tristan_display_name.unwrap(), tristan_on_mission.unwrap(), iseult_display_name, iseult_on_mission));
+        }
+        // Next do the same for Iseult.
+        if iseult_display_name.is_some() {
+            effects.push(self.build_lover_effect(&iseult_display_name.unwrap(), iseult_on_mission.unwrap(), tristan_display_name, tristan_on_mission));
+        }
+    }
+
+    // Builds a message for a player, given information about their lover who may not exist.
+    fn build_lover_effect(&self, player_display_name: &str, player_on_mission: bool, opt_lover_display_name: Option<&String>, opt_lover_on_mission: Option<bool>) -> Effect {        
+        let inner_message = opt_lover_on_mission.map_or("does not exist. You're alone :(".to_owned(), 
```

Comment:
nit: Can the `:(` be a `:'(`?

Reply:
I clearly underestimated the sadness of being a lone lover